packages/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js

```diff
@@ -129,20 +130,32 @@ class KeyboardAvoidingView extends React.Component<Props, State> {
     }
   };
 
+  // Avoid unnecessary renders if the KeyboardAvoidingView is disabled.
+  _setBottom = (value: number) => {
+    const {enabled = true} = this.props;
+    this._bottom = value;
+    if (enabled === true) {
+      this.setState({bottom: value});
+    }
+  };
+
   _updateBottomIfNecessary = async () => {
     if (this._keyboardEvent == null) {
-      this.setState({bottom: 0});
+      this._setBottom(0);
       return;
     }
 
     const {duration, easing, endCoordinates} = this._keyboardEvent;
     const height = await this._relativeKeyboardHeight(endCoordinates);
 
-    if (this.state.bottom === height) {
+    if (this._bottom === height) {
       return;
     }
 
-    if (duration && easing) {
+    this._setBottom(height);
+
+    const {enabled = true} = this.props;
+    if (enabled === true && duration && easing) {
```

Comment:
```suggestion
    if (enabled && duration && easing) {
```
Self explanatory, less verbose :)